### ISSUE TITLE
fork command line util argument for path to open

### ISSUE THREAD

**Issue by danh1979** [ForkIssues/Tracker#6]
I like using the `fork` command line tool to launch a Fork window from a terminal within a repository, but I would like to be able to give it the path to the repository I would like it to open.

e.g. if I am in a directory (perhaps a super-repo!) with multiple repos, I'd like to type only
```
fork repo-foo
fork repo-bar
```
instead of
```
pushd repo-foo ; fork ; popd
pushd repo-bar ; fork ; popd
```

Ideally for me, `fork` and `Fork.app` would work somewhat more like how the `mate` command line tool and `TextMate.app` work together with multiple terminals and windows across multiple desktops.

**Comment by DanPristupov** [comment]
Currently you can use the `--git-dir` option to call Fork without leaving the current directory:
```
fork --git-dir=repo-foo
fork log --git-dir=repo-bar
```

I'm thinking of adding an additional short parameter -C which will do the same.

**Comment by danh1979** [comment]
Thanks, `--git-dir=/path/to/dir` is useful, and I'd welcome `-C`. I appreciate that they are the same options as `git` itself. Perhaps add this information to the output of `--help`?

As I most often quickly launch `Fork.app` via `fork` from whatever terminal I happen to be working in, I feel it is extra boilerplate typing, enough to have created a work-around:
```bash
#!/usr/bin/env bash
fork=/usr/local/bin/fork
args=()
git_dir="."
for arg in $@
do
    case "${arg}" in 
    "--help"|"--version"|open|status|commit|log) # known fork options go here
        args+=("${arg}")
        ;;
    *) # else assume it is a path to the git dir
        git_dir="${arg}"
        ;;
    esac
done
exec ${fork} ${args[@]} --git-dir="${git_dir}"
```
with an alias in my `.bash_profile`:
```bash
alias fork="~/bin/fork.sh"
```

Unexpectedly, I can only get `fork` to switch to the commit view if the `--git-dir=path/to/foo` is last.
e.g.
```
/usr/local/bin/fork --git-dir=path/to/repo commit
```
does *not* switch to commit view (if it is not already active), whereas
```
/usr/local/bin/fork commit --git-dir=path/to/repo
```
does.

If I was designing the `fork` tool from scratch for my own needs, it would probably look like:
```
$ fork --help
Fork Command Line Tools v42.
usage: fork [-s|-c|-l] [PATH]

    -c        switch to Commit view
    -s        switch to Status view (same as -c)
    -l        switch to Revision List view

    optional PATH argument defaults to current directory
````

**Comment by DanPristupov** [comment]
Actually my intention was to not implement a Fork CLI syntax, but use the git's one instead. In this way users don't need to read help and/or memorize commands in order to make simple things.

Fork CLI uses the following pattern: `fork command --options`.

Currently:

```
#show the commit log
git  log
fork log

# show repository status
git  status
fork status

# make commit
git  commit
fork commit # should show the commit view and focus the message field

```

In plans:

```
# show diff between two commits
git  diff HEAD origin/master
fork diff sha1..sha2

# show commit log for a branch
git  log master
fork log master

# or all commits
git  log --all
fork log --all

# show a file history
git  log -- /path/to/file
fork log -- /path/to/file

# merge a branch into HEAD
git  merge develop
fork merge develop

...
```

This implies the following:
1. The commands don't have short forms (same as in git). Options can have short alternatives.
2. Command options (including --git-dir) must follow the command.

Does it make sense? It would appreciate any feedback regarding such approach.

**Comment by aziz** [comment]
Following the git cli pattern is a great idea.

**Comment by danh1979** [comment]
I agree that reusing the git cli pattern is a good idea in a cognitive load sense. `fork -C path/to/repo` might be enough typing reduction for me to drop my alias.

Thanks for sharing your plans.

**Comment by danielfrey** [comment]
Great idea! Especially `fork log -- /path/to/file` would be outstanding! It would allow a simple integration to call it from an editor.

**Comment by DanPristupov** [comment]
@danielfrey Fork also needs working directory to be configured.

So integration in an editor probably will require something more complicated. Like following:
`fork -C /absolute/path/to/repository log -- relative/path/to/file`

Explanation:
`-C /path/` - define the repository directory. It's better to use absolute path here (e.g.`/Users/username/sources/myrepo`)
`log -- relative/path/to/file` - define the file path relatively to repository

**Comment by DanPristupov** [comment]
@danh1979 just released Fork 1.0.54 which supports the `fork log [-- path/to/file]` cli command.

**Comment by danielfrey** [comment]
There seems to be a minor bug: 
If I call `fork log -- absolute path` it opens the application, inside a popup appears. So far so good. 
But if I want to close it again with the close-button (or ESC), I need three attempts or more. The closed popup reappears several times until it finally quits. 
The feature itself it great! 😃

**Comment by DanPristupov** [comment]
@danh1979 ouch, sorry. I had to test it more. Will fix in the next version.
